internal/ethapi/api.go

```diff
@@ -848,6 +870,14 @@ func (s *PublicBlockChainAPI) GetBlockByHash(ctx context.Context, hash common.Ha
 		response, err := s.rpcMarshalBlock(ctx, block, true, fullTx)
 		// append marshalled bor transaction
 		if err == nil && response != nil {
+
+			author, err := s.getAuthor(block.Header())
+			if err != nil {
+				return nil, err
```

Comment:
Do we want to return an error from here or just return the block with `0x00` miner address?

Reply:
I agree. What I mean to say is `getBlockByNumber` is highly used function. Any errors with our `getAuthor` implementation should not ideally cause error in this function. We're coupling these 2 components for this.

Reply:
> I agree. What I mean to say is `getBlockByNumber` is highly used function. Any errors with our `getAuthor` implementation should not ideally cause error in this function. We're coupling these 2 components for this.

I agree.

Reply:
good point manav
ideally the functionality should not change of the public facing function
if we are able to find the value from getAuthor, it should return the value otherwise it should return "0x0000..." (previous functionality)

Reply:
Okay, will do that change rn.